### PR TITLE
fix: correct github oauth2 callback url

### LINKED ISSUE #9034: Unable to change login type to GitHub
I've just gotten Coder OSS set up on my Kubernetes cluster. I enabled GitHub auth, but since I used the same email address as my GItHub account to create my account originally, it told me to change login type in my account settings. I attempted to do that, and it presented me with a confirmation dialog where I entered my password, but then it took me to a blank page with the error:

```
{"message":"\"user\" must be an existing uuid or username."}
```

_Originally posted by @ryansburgoyne in https://github.com/coder/coder/discussions/9031_

### PR DESCRIPTION
**Pull request by Emyrk**
URL was missing `oauth2` part of the path.

Fixes: https://github.com/coder/coder/issues/9034#issuecomment-1673863805

### CODE REVIEW

File: site/src/pages/UserSettingsPage/SecurityPage/SingleSignOnSection.tsx [hunk 1]
```diff
@@ -30,7 +30,14 @@ export const redirectToOIDCAuth = (
   stateString: string,
   redirectTo: string,
 ) => {
-  window.location.href = `/api/v2/users/${toType}/callback?oidc_merge_state=${stateString}&redirect=${redirectTo}`
+  switch (toType) {
+    case "github":
+      return (window.location.href = `/api/v2/users/oauth2/github/callback?oidc_merge_state=${stateString}&redirect=${redirectTo}`)
```
Comment (deansheather) [inline site/src/pages/UserSettingsPage/SecurityPage/SingleSignOnSection.tsx:35]: It's kinda weird to return an assignment statement, what about assigning and then returning on the next line?

**Review by deansheather — APPROVED** [review]

File: site/src/pages/UserSettingsPage/SecurityPage/SingleSignOnSection.tsx (hunk 1 above)
Comment (Emyrk) [reply site/src/pages/UserSettingsPage/SecurityPage/SingleSignOnSection.tsx:35]: Will do that :+1:

Comment (Emyrk) [reply site/src/pages/UserSettingsPage/SecurityPage/SingleSignOnSection.tsx:35]: tbh I wasn't sure if I even needed a "return", but my IDE did that so the switch statements do not fallthrough

Comment (Emyrk) [reply site/src/pages/UserSettingsPage/SecurityPage/SingleSignOnSection.tsx]: I just added breaks to break the switch statement. No returns

**Pull request merged**